### ISSUE TITLE
xcatd on service node does not work due to missing "perl-DBD-MySQL" 

### ISSUE THREAD

**Issue by immarvin** [xcat2/xcat-core#5728]
this is a issue reported in mail list,

>I have deployed a service node, to the best of my knowledge, using the online documentation.  I've decided to rsync the install and tftpboot folders to deploy nodes directly from it.  rpm -qa shows the service node rpm was applied 
>Attempting to start the xcatd daemon on the service node hangs, the errors being 
>"Oct 18 16:22:18 service01 xcat[7109]:  Could not connect to the database. Database handle not defined."

```
[root@service01 ~]# XCATBYPASS=1 lsxcatd -a
Could not connect to the database. Database handle not defined.
 Could not read the site table
 
Could not connect to the database. Database handle not defined.
 Could not read the site table
 
Could not connect to the database. Database handle not defined.
 Could not read the site table
 
Could not connect to the database. Database handle not defined.
 Could not read the site table
 
Could not connect to the database. Database handle not defined.
 Could not read the site table
 
Could not connect to the database. Database handle not defined.
 Could not read the site table
 
Could not connect to the database. Database handle not defined.
 Could not read the site table
 
Could not connect to the database. Database handle not defined.
 Could not read the site table
 
Could not connect to the database. Database handle not defined.
 Could not read the site table
 
Could not connect to the database. Database handle not defined.
Error loading module /opt/xcat/lib/perl/xCAT_plugin/zvmdiscovery.pm  ...skipping
Could not connect to the database. Database handle not defined.
ERROR: Unable to open basic site table for configuration
Version 2.14.1 (git commit 70d6e7f93cc9714a127c22df2e7ca53d4996a34c, built Fri Jun  1 03:00:53 EDT 2018)
This is a Service Node
cfgloc=mysql:dbname=xcatdb;host=10.4.0.1|xcatadmin
dbengine=mysql
dbname=xcatdb
dbhost=10.4.0.1
dbadmin=xcatadmin
 ```

finally, user find the root cause by themselves:

>There was a missing package he noticed in the strace output for xcatd.  There was a file mentioned over and over, he did a lookup for the package that provides the file 
```
(stat("/usr/local/share/perl5/DBD/mysql.pmc", 0x7ffcec9af410) = -1 ENOENT (No such file or directory)
(stat("/usr/local/share/perl5/DBD/mysql.pm", 0x7ffcec9af350) = -1 ENOENT (No such file or directory)   
```
> The missing package was:  perl-DBD-MySQL
> Once I installed that, xcatd started and I can run lsdef and other xCAT commands.  If I missed this from the documentation, a thousand pardons :)  However, if this is a bug with the service node package list, it needs to be remedied by the devs.

 currently,  `perl-DBD-MySQL ` is not specified in xCATsn.spec, it is specified in the service node osimage instead:

```
[root@c910f03c05k21 xcat-core]# grep perl-DBD-MySQL ./* -R
./docs/source/advanced/hierarchy/databases/mysql_install.rst:       perl-DBD-MySQL*
./docs/source/advanced/hierarchy/databases/mysql_install.rst:       perl-DBD-MySQL*
./xCAT-server/share/xcat/install/fedora/service.ppc64.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/install/fedora/service.fedora9.ppc64.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/install/fedora/service.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/install/fedora/service.fedora9.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/install/rh/service.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/install/rh/service.rhels6.x86_64.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/install/rh/service.rhels7.x86_64.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/install/rh/service.rhels7.ppc64.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/install/rh/service.rhels7.ppc64le.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/install/rh/service.rhels6.ppc64.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/install/rh/service.rhels6.ppc64.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/install/rh/service.ppc64.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/netboot/fedora/service.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/netboot/rh/service.ppc64.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/netboot/rh/service.rhels6.ppc64.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/netboot/rh/service.rhels7.ppc64le.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/netboot/rh/service.rhels7.x86_64.pkglist:perl-DBD-MySQL
./xCAT-server/share/xcat/netboot/rh/service.pkglist:perl-DBD-MySQL
```

**Comment by cxhong** [comment]
`perl-DBD-MySQL` is included in the service pkglist.  when you deploy a service node , you should use the `service` osimage.   In the following case, sn02 is a service node
````
# lsdef sn02 -i provmethod
Object name: sn02
    provmethod=rhels7.5-alternate-ppc64le-install-service
# lsdef -t osimage rhels7.5-alternate-ppc64le-install-service -i pkglist
Object name: rhels7.5-alternate-ppc64le-install-service
    pkglist=/opt/xcat/share/xcat/install/rh/service.rhels7.ppc64le.pkglist
# grep perl-DBD /opt/xcat/share/xcat/install/rh/service.rhels7.ppc64le.pkglist
perl-DBD-MySQL
perl-DBD-Pg
````
So, please make sure you are using service osimage and service pkglist contains `perl-DBD-MySQL`

**Comment by gurevichmark** [comment]
Looks like this was fixed in #6234
